src/entities/inputValidator/inputValidator.ts

```diff
@@ -0,0 +1,49 @@
+import { AddLiquidityInput } from '../addLiquidity';
+import { CreatePoolInput } from '../createPool/types';
+import { PoolStateInput } from '../types';
+import { InputValidatorComposableStable } from './composableStable/inputValidatorComposableStable';
+import { InputValidatorGyro } from './gyro/inputValidatorGyro';
+import { InputValidatorBase } from './types';
+import { InputValidatorWeighted } from './weighted/inputValidatorWeighted';
+
+export class InputValidator {
+    validators: Record<string, InputValidatorBase> = {};
```

Comment:
I know this replicates how other classes are implementing this, but I think we could benefit from using a PoolType type instead of string. This can be done by altering the api-client to map poolType from `string` to `PoolType`. I'll create a task to evaluate this refactor in the future. As reference, Add/Remove Nested Liquidity PR already has this change in place.

Reply:
Yeah, I used it as string because I didn't change the PoolType class to UpperCase, api sends the type as WEIGHTED, PHANTOM_STABLE, GYRO2... etc, I can create an adapter to bring the returned pool types from API as an enum PoolType value, what do you think?

Reply:
I'd say it's ok to leave as is for now - I just created #157 so we don't forget to come back to this in the future.